### PR TITLE
Adding loadbalancer support to java service instance

### PR DESCRIPTION
**Pull request by mbfrahry**
This PR adds support for `load_balancer` to java service instance. The vendored version of Terraform needed be updated to v0.11.x to take advantage of d.GetOkExists. 

```
=== RUN   TestAccOraclePAASJavaServiceInstance_LoadBalancer
--- PASS: TestAccOraclePAASJavaServiceInstance_LoadBalancer (2186.97s)
```

### CODE REVIEW

File: oraclepaas/resource_java_service_instance.go [hunk 1]
```diff
@@ -789,6 +839,12 @@ func resourceOraclePAASJavaServiceInstanceRead(d *schema.ResourceData, meta inte
 		d.Set("assign_public_ip", val)
 	}
 
+	if result.LoadBalancer != nil {
+		if err := d.Set("load_balancer", flattenLoadBalancer(d, *result.LoadBalancer)); err != nil {
```
Comment (tombuildsstuff) [inline oraclepaas/resource_java_service_instance.go]: by nil-checking this we're not allowing this to be removed; which whilst technically correct given this is ForceNew, in general it's better to allow flattenLoadBalancer to take a pointer and then return an empty item if it's nil imo

Comment (mbfrahry) [reply oraclepaas/resource_java_service_instance.go]: Ah, that makes sense to me. Fixed up now

File: oraclepaas/resource_java_service_instance.go [hunk 2]
```diff
@@ -1228,8 +1226,11 @@ func expandWLSPorts(webLogicServer *java.CreateWLS, config map[string]interface{
 	}
 }
 
-func flattenLoadBalancer(d *schema.ResourceData, loadBalancerInfo java.LoadBalancerInfo) []interface{} {
+func flattenLoadBalancer(d *schema.ResourceData, loadBalancerInfo *java.LoadBalancerInfo) []interface{} {
 	result := make(map[string]interface{})
+	if loadBalancerInfo != nil {
+		return []interface{}{result}
```
Comment (tombuildsstuff) [inline oraclepaas/resource_java_service_instance.go]: this wants to be:

```
if loadBalancerInfo == nil {
  return []interface{}{}
}
```

File: oraclepaas/resource_java_service_instance.go [hunk 3]
```diff
@@ -645,6 +646,11 @@ func resourceOraclePAASJavaServiceInstance() *schema.Resource {
 				Default:  false,
 				ForceNew: true,
 			},
+			"use_oauth_for_storage": {
```
[the remark this answers is not in the document]
Comment (mbfrahry) [reply oraclepaas/resource_java_service_instance.go]: Moved!

**Comment by aorcl** [comment]
Thank you everybody!

**Pull request merged**